src/main/java/pl/llp/aircasting/screens/stream/NoteViewerActivity.java

```diff
@@ -0,0 +1,157 @@
+        mNoteDelete = findViewById(R.id.note_delete);
+        mNoteSave = findViewById(R.id.note_save);
+
+        mNextNote.setOnClickListener(this);
+        mPreviousNote.setOnClickListener(this);
+        mViewPhoto.setOnClickListener(this);
+        mNoteSave.setOnClickListener(this);
+        mNoteDelete.setOnClickListener(this);
+
+        mNotesTotal = mVisibleSession.getSessionNoteCount();
+        mNoteIndex = getIntent().getIntExtra(NOTE_INDEX, 0);
+
+        setNote();
+    }
+
+    private void setNote() {
+        int index = ((mNoteIndex % mNotesTotal) + mNotesTotal) % mNotesTotal;
+        mCurrentNote = mVisibleSession.getSessionNote(index);
+        String title = FormatHelper.dateTime(mCurrentNote.getDate()).toString();
+        mNoteDate.setText(title);
+
+        mNoteNumber.setText(index + 1 + "/" + mNotesTotal);
+        mNoteText.setText(mCurrentNote.getText());
+        mViewPhoto.setVisibility(mPhotoHelper.photoExists(mCurrentNote) ? View.VISIBLE : View.GONE);
+
+        mNoteIndex = index;
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()) {
+            case R.id.note_left:
+                mNoteIndex -= 1;
+                setNote();
+                break;
+            case R.id.note_right:
+                mNoteIndex += 1;
+                setNote();
+                break;
+            case R.id.view_photo:
+                showNotePhoto();
+                break;
+            case R.id.note_save:
+                saveNote();
+                break;
+            case R.id.note_delete:
+                deleteNote();
+                break;
+        }
+    }
+
+    private void saveNote() {
+        String text = mNoteText.getText().toString();
+        mCurrentNote.setText(text);
+        final Context context = this;
+
+        new SimpleProgressTask<Void, Void, Void>(this) {
+            @Override
+            protected Void doInBackground(Void... voids) {
+                if (mVisibleSession.isVisibleSessionViewed()) {
+                    mNotesRepository.updateNote(mCurrentNote, mVisibleSession.getSession().getId());
+                    triggerSync(context);
+                }
+                return null;
+            }
+        }.execute();
+
+        finish();
+    }
+
+    private void deleteNote() {
+        final Context context = this;
+
+        new SimpleProgressTask<Void, Void, Void>(this) {
+            @Override
+            protected Void doInBackground(Void... voids) {
+                mNotesRepository.deleteNote(mVisibleSession.getSession().getId(), mCurrentNote.getNumber());
+                mVisibleSession.deleteNote(mCurrentNote);
+                triggerSync(context);
+
+                return null;
+            }
+        }.execute();
+
+        finish();
+    }
+
+    private void showNotePhoto() {
+        Intents.viewPhoto(this, photoUri());
+    }
+
+    private Uri photoUri() {
+        if (mPhotoHelper.photoExistsLocally(mCurrentNote)) {
```

Comment:
Could you elaborate on what's the if checking?

Reply:
We are checking if the file exists locally, on the device. Because most of the time it won't, it will only be a url to a photo saved in the backend. These are different cases and they can be opened in different ways. This is actually what I'm working on right now, there's a problem with local photos not displaying, while the ones coming from the server are fine.